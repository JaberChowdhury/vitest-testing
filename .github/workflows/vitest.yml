name: Run Vitest Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      # Step 1: Check out the repository
      - name: Checkout code
        uses: actions/checkout@v3

      # Step 2: Set up Node.js with pnpm caching
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: "pnpm"

      # Step 3: Install pnpm manually (bypassing Corepack)
      - name: Install pnpm
        run: npm install -g pnpm

      # Step 4: Ensure pnpm is added to PATH
      - name: Add pnpm to PATH
        run: echo "$(npm root -g)/pnpm" >> $GITHUB_PATH

      # Step 5: Verify pnpm installation and check PATH
      - name: Verify pnpm installation
        run: |
          echo "pnpm version:"
          pnpm --version
          echo "pnpm location:"
          which pnpm
          echo "Current PATH:"
          echo $PATH

      # Step 6: Install dependencies
      - name: Install dependencies
        run: pnpm install

      # Step 7: Run Vitest tests
      - name: Run tests
        run: pnpm test
