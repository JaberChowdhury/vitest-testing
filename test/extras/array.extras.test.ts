import { describe, expect, test } from "vitest";

describe("Array extras", () => {
  // Array Length
  test("array has correct length", () => {
    const arr = [1, 2, 3];
    expect(arr).toHaveLength(3);
  });
});

describe("Array extras", () => {
  // Array Contains an Item
  test("array contains an item", () => {
    const arr = ["apple", "banana", "cherry"];
    expect(arr).toContain("banana");
  });
});
describe("Array extras", () => {
  // Array Does Not Contain an Item
  test("array does not contain an item", () => {
    const arr = ["apple", "banana", "cherry"];
    expect(arr).not.toContain("grape");
  });
});

describe("Array extras", () => {
  // Array Contains All Items
  test("array contains all items", () => {
    const arr = [1, 2, 3, 4, 5, 7];
    expect(arr).toEqual(expect.arrayContaining([2, 5]));
  });
});
describe("Array extras", () => {
  // Array Structure Matching
  test("array structure matches", () => {
    const users = [
      { id: 1, name: "Alice" },
      { id: 2, name: "Bob" },
    ];
    expect(users).toEqual([
      { id: 1, name: "Alice" },
      { id: 2, name: "Bob" },
    ]);
  });
});
describe("Array extras", () => {
  //  Empty Array
  test("array is empty", () => {
    const arr = [];
    expect(arr).toHaveLength(0);
    expect(arr).toEqual([]);
  });
});
describe("Array extras", () => {
  // Array Sorting
  test("array is sorted", () => {
    const arr = [1, 2, 3, 4, 5];
    const isSorted = arr.every((val, i, arr) => !i || val >= arr[i - 1]);
    expect(isSorted).toBeTruthy();
  });
});
describe("Array extras", () => {
  // Snapshot Testing for Arrays
  test("array snapshot", () => {
    const arr = [1, 2, 3];
    expect(arr).toMatchSnapshot();
  });
});
