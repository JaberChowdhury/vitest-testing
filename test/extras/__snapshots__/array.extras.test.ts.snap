// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Array extras > array snapshot 1`] = `
[
  1,
  2,
  3,
]
`;
